html/includes/api_functions.inc.php

```diff
@@ -51,6 +51,7 @@ function get_graph_by_port_hostname()
     $app          = \Slim\Slim::getInstance();
     $router       = $app->router()->getCurrentRoute()->getParams();
     $hostname     = $router['hostname'];
+	$device_id    = ctype_digit($hostname) ? $hostname : getidbyname($hostname);
```

Comment:
I'm not sure that it matters, but this does an additional sql query.
You could also modify the query below.